src/main.rs

```diff
@@ -111,8 +111,10 @@ fn get_env(key: &str) -> anyhow::Result<String> {
 fn main() {
     init_log();
     if let Err(err) = app() {
-        error!("{}", err);
-        for cause in err.chain() {
+        // Pull off the first element of the chain as the first element. The chain contains all the
+        // elements, not just the causes.
+        error!("failed: {}", err.chain().next().unwrap());
```

Comment:
Hm, perhaps that would work too. I find Display impl to be a little more confusing though (less obvious that it pulls the first element, and so that we do get all the elements).

I'll merge this for now to avoid another review bounce since this is blocking prod though.